populus/compilation/compile.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import os
+
+from .helpers import (
+    get_dir_source_paths,
+    post_process_compiled_contracts,
+    validate_compiled_contracts,
+)
+
+from .backends.solc_auto import (
+    get_compiler_backend_class_for_version
+)
+
+from populus.utils.functional import (
+    get_duplicates,
+)
+
+
+from populus.config.loading import (
+    load_global_config,
+)
+
+
+def _get_contract_key(contract_data):
+    return (
+        contract_data['source_path'],
+        contract_data['name'],
+    )
+
+
+def compile_dirs(dir_paths,
+                 global_config_path=None,
+                 import_remappings=None,
+                 compiler_version="auto"
+                 ):
+
+    logger = logging.getLogger('populus.compilation.helpers.compile_dir')
+    if import_remappings is None:
+        import_remappings = []
+
+    all_source_paths = []
+    for dir_path in set(dir_paths): # sliently ignore duplicates
```

Comment:
I think that we can enforce this uniqueness in the JSON schema.  Can you look into that.

Similarly, there is an `is_same_path` function that would allow us to detect duplicates that are not exact string matches.  It seems like something where if we do allow duplicates, we should still potentially issue a warning that a path was found multiple times and that the duplicate should be removed.